Sources/UnidirectionalBinding.swift

```diff
@@ -304,14 +305,17 @@ public final class BindingTarget<Value>: BindingTargetProtocol {
 	///   - setter: The action to receive values.
 	public convenience init(on queue: DispatchQueue, lifetime: Lifetime, setter: @escaping (Value) -> Void) {
 		let queueId = ObjectIdentifier(queue)
+		let usesMainQueue = queue == DispatchQueue.main
 
 		/// Ensures the queue has been setup property.
 		if nil == queue.getSpecific(key: specificKey) {
 			queue.setSpecific(key: specificKey, value: queueId)
 		}
 
 		let setter: (Value) -> Void = { value in
-			if queueId == DispatchQueue.getSpecific(key: specificKey) {
+			if usesMainQueue && Thread.isMainThread {
```

Comment:
What if we instead did this, and killed the `Bool` variable?

```
if queue == DispatchQueue.main && Thread.isMainThread {
```

That, or `s/usesMainQueue/onMainQueue` so it communicates the intent a little more clearly. I know I'm splitting hairs here, but nobody's really *using* the main queue.